crates/wasmtime/src/runtime/vm/sys/unix/signals.rs

```diff
@@ -347,6 +347,12 @@ unsafe fn get_trap_registers(cx: *mut libc::c_void, _signum: libc::c_int) -> Tra
                 pc: cx.sc_rip as usize,
                 fp: cx.sc_rbp as usize,
             }
+        } else if #[cfg(all(target_os = "illumos", target_arch = "x86_64"))] {
+            let cx = &*(cx as *const libc::ucontext_t);
+            TrapRegisters {
+                pc: cx.uc_mcontext.gregs[libc::REG_RIP as usize] as usize,
+                fp: cx.uc_mcontext.gregs[libc::REG_RBP as usize] as usize,
```

Comment:
on second read, mind moving this up to the first case? I think this is the same as that up there and should only need an update to the `#[cfg]` to include illumos too

Reply:
Ah good catch.